firewood/src/proofs/diff.rs

```diff
@@ -0,0 +1,2028 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+// TODO: Currently only tests use the code in this file. Remove allow(dead_code) once change
+//       proof has been implemented.
+#![allow(dead_code)]
```

Comment:
This is likely to remain after we start using it, since it's not a `expect(dead_code)`. Also, it's better to just put this on the top level function it's complaining about, rather than the whole file, in case there's something stray you're not using.

Moving this to `#[expect(dead_code)]` to `DiffMerkleNodeStream` probably solves both of these.